datalad_next/shell/__init__.py

```diff
@@ -123,6 +123,12 @@ class identified by ``zero_command_rg_class`` will be used by default to create
 per-call basis by providing a different response generator class in the
 ``response_generator``-parameter of :meth:`ShellCommandExecutor.__call__`.
 
+Examples
+--------
+
+See the documentation of :func:`datalad_next.shell.shell` for examples of how to
+use the shell-function and different response generator classes.
+
```

Comment:
Adding a subheading here. Otherwise the API table below makes the impression to be a list of examples.

```suggestion

API overview
------------
```